java/code/src/com/redhat/rhn/frontend/xmlrpc/contentmgmt/ContentManagementHandler.java

```diff
@@ -0,0 +1,271 @@
+     * @xmlrpc.param #param_desc("string", "name", "new Content Environment name")
+     * @xmlrpc.param #param_desc("string", "description", "new Content Environment description")
+     * @xmlrpc.returntype $ContentEnvironmentSerializer
+     */
+    public ContentEnvironment createEnvironment(User loggedInUser, String projectLabel, String predecessorLabel,
+            String label, String name, String description) {
+        ensureOrgAdmin(loggedInUser);
+        try {
+            return ContentManager.createEnvironment(projectLabel, of(predecessorLabel), label, name, description,
+                    loggedInUser);
+        }
+        catch (ContentManagementException e) {
+            throw new ContentManagementException(e.getMessage());
+        }
+    }
+
+    /**
+     * Update Content Environment
+     *
+     * @param loggedInUser - the logged in user
+     * @param projectLabel - the Content Project label
+     * @param envLabel - the Environment label
+     * @param props - the map with the Environment properties
+     * @return the updated Environment
+     *
+     * @xmlrpc.doc Update Content Environment with given label
+     * @xmlrpc.param #session_key()
+     * @xmlrpc.param #param_desc("string", "projectLabel", "Content Project label")
+     * @xmlrpc.param #param_desc("string", "envLabel", "Content Environment label")
+     * @xmlrpc.param
+     *  #struct("data")
+     *      #prop_desc("string", "name", "Content Environment name")
+     *      #prop_desc("string", "description", "Content Environment description")
+     *  #struct_end()
+     * @xmlrpc.returntype $ContentEnvironmentSerializer
+     */
+    public ContentEnvironment updateEnvironment(User loggedInUser, String projectLabel, String envLabel,
+            Map<String, Object> props) {
+        ensureOrgAdmin(loggedInUser);
+        try {
+            return ContentManager.updateEnvironment(envLabel,
+                    projectLabel,
+                    ofNullable((String) props.get("name")),
+                    ofNullable((String) props.get("description")),
+                    loggedInUser);
+        }
+        catch (ContentManagementException e) {
+            throw new InvalidParameterException(e.getMessage());
+        }
+    }
+
+    /**
+     * Remove a Content Environment
+     *
+     * @param loggedInUser - the logged in user
+     * @param projectLabel - the Content Project label
+     * @param envLabel - the Content Environment label
+     * @return the number of removed objects
+     *
+     * @xmlrpc.doc Remove a Content Environment
+     * @xmlrpc.param #session_key()
+     * @xmlrpc.param #param_desc("string", "projectLabel", "Content Project label")
+     * @xmlrpc.param #param_desc("string", "envLabel", "Content Environment label")
+     * @xmlrpc.returntype int - the number of removed objects
+     */
+    public int removeEnvironment(User loggedInUser, String projectLabel, String envLabel) {
+        ensureOrgAdmin(loggedInUser);
```

Comment:
I think here we are going to throw the other exception because this class extends the `BaseHelper` as per *my other comment*

Reply:
(Not sure if I get this comment correctly, please correct anything below: :) )

If we were throwing that `PermissionCheckFailureException` from the `ContentManager` method, we wouldn't need to check it here, that's right. But as I wrote, I think it's better to save the xmlrpc exceptions fort he xmlrpc package.

What we alternatively could do here is to catch that `PermissionException` here and rethrow the `PermissionCheckFailureException`. But I wanted to stay consistent with other handlers and explicitly put the `ensure...` method there. It's immediately clear that this endpoint is not usable without `ORG_ADMIN` role.

Reply:
Definitely agreed, `consistency` wins over everything else, :wink: